test/foundry/dapp/Application.t.sol

```diff
@@ -425,6 +426,78 @@ contract ApplicationTest is ERC165Test {
         _appContract.executeOutput(output, proof);
     }
 
+    function testExecuteERC20TransferDelegateCallVoucher() external {
+        string memory name = "ERC20DelegateCallVoucher";
+        bytes memory output = _getOutput(name);
+        OutputValidityProof memory proof = _getProof(name);
+
+        assertLt(
+            _erc20Token.balanceOf(address(_appContract)),
+            _transferAmount,
+            "Application contract does not have enough ERC-20 tokens"
+        );
+
+        // test revert
+
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                IERC20Errors.ERC20InsufficientBalance.selector,
+                address(_appContract),
+                _erc20Token.balanceOf(address(_appContract)),
+                _transferAmount
+            )
+        );
+        _appContract.executeOutput(output, proof);
+
+        // fund dapp
```

Comment:
It's self-evident that we're funding the application contract, so we don't need to comment that.